forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestFactory.java

```diff
@@ -0,0 +1,159 @@
+                request.setJson(registrationRequestJson);
+                request.setDirectorySoftwareStatement(regRequestSoftwareStatement);
+                return request;
+            } catch (IOException ioe){
+
+                throw new DynamicClientRegistrationException("Could not map registration request jwt fields to " +
+                        "internal object " + ioe.getMessage(), DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA);
+            }
+        } catch (ParseException pe){
+            String errorMessage = "Could not parse the provided registration JWT";
+            log.debug("getRegistrationRequest() {}", errorMessage, pe);
+            throw new DynamicClientRegistrationException(errorMessage,
+                    DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA );
+        }
+    }
+
+
+    private RegistrationRequestJWTClaims getJwtClaimsSet(String registrationRequestJwtSerialised,
+                                                         JWTClaimsOrigin origin) throws ParseException,
+            DynamicClientRegistrationException {
+        SignedJWT registrationRequestJws = SignedJWT.parse(registrationRequestJwtSerialised);
+        JWTClaimsSet registrationRequestClaimsSet = registrationRequestJws.getJWTClaimsSet();
+        if(registrationRequestClaimsSet == null){
+            String errorString = "Registration Request JWT contained no claims";
+            log.debug("getRegistrationRequestFromJwt() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString,
+                    DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA);
+        }
+        RegistrationRequestJWTClaims registrationRequestJWTClaims =
+                new RegistrationRequestJWTClaims(registrationRequestClaimsSet,
+                        origin);
+        return registrationRequestJWTClaims;
+    }
+
+    public RegistrationRequest getRegistrationRequestFromManualRegistrationJson(String registrationRequestJson,
+                                                                                ManualRegistrationRequest manualRegistrationRequest,
+                                                                                ObjectMapper objectMapper)
+            throws DynamicClientRegistrationException {
+        try{
+            RegistrationRequest request = objectMapper.readValue(registrationRequestJson, RegistrationRequest.class);
+            request.setRedirectUris(manualRegistrationRequest.getRedirectUris());
+            String softwareStatementAssertion = manualRegistrationRequest.getSoftwareStatementAssertion();
+            if (StringUtils.isEmpty(softwareStatementAssertion)){
+                String errorMessage = "Manual Request did not contain a valid software statement";
+                log.info("getRegistrationRequestFromManualregistrationJson() {}", errorMessage);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.